src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -1,73 +1,24 @@
-        internal const int DefaultBufferSize = 4096;
-
-        private byte[]? _buffer;
-        private int _bufferLength;
-        private readonly SafeFileHandle _fileHandle; // only ever null if ctor throws
-
-        /// <summary>Whether the file is opened for reading, writing, or both.</summary>
-        private readonly FileAccess _access;
-
-        /// <summary>The path to the opened file.</summary>
-        private readonly string? _path;
-
-        /// <summary>The next available byte to be read from the _buffer.</summary>
-        private int _readPos;
-
-        /// <summary>The number of valid bytes in _buffer.</summary>
-        private int _readLength;
-
-        /// <summary>The next location in which a write should occur to the buffer.</summary>
-        private int _writePos;
-
-        /// <summary>
-        /// Whether asynchronous read/write/flush operations should be performed using async I/O.
-        /// On Windows FileOptions.Asynchronous controls how the file handle is configured,
-        /// and then as a result how operations are issued against that file handle.  On Unix,
-        /// there isn't any distinction around how file descriptors are created for async vs
-        /// sync, but we still differentiate how the operations are issued in order to provide
-        /// similar behavioral semantics and performance characteristics as on Windows.  On
-        /// Windows, if non-async, async read/write requests just delegate to the base stream,
-        /// and no attempt is made to synchronize between sync and async operations on the stream;
-        /// if async, then async read/write requests are implemented specially, and sync read/write
-        /// requests are coordinated with async ones by implementing the sync ones over the async
-        /// ones.  On Unix, we do something similar.  If non-async, async read/write requests just
-        /// delegate to the base stream, and no attempt is made to synchronize.  If async, we use
-        /// a semaphore to coordinate both sync and async operations.
-        /// </summary>
-        private readonly bool _useAsyncIO;
-
-        /// <summary>cached task for read ops that complete synchronously</summary>
-        private Task<int>? _lastSynchronouslyCompletedTask;
-
-        /// <summary>
-        /// Currently cached position in the stream.  This should always mirror the underlying file's actual position,
-        /// and should only ever be out of sync if another stream with access to this same file manipulates it, at which
-        /// point we attempt to error out.
-        /// </summary>
-        private long _filePosition;
-
-        /// <summary>Whether the file stream's handle has been exposed.</summary>
-        private bool _exposedHandle;
 
         /// <summary>Caches whether Serialization Guard has been disabled for file writes</summary>
         private static int s_cachedSerializationSwitch;
 
+        private readonly FileStreamStrategy _impl;
```

Comment:
Nit: If we're going to name the field _impl and if concrete implementations are going to end in Impl, it's strange for the base type to be named Strategy.  It should instead be Base or ImplBase or something like that.